plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/PropertiesSecretSource.java

```diff
@@ -0,0 +1,61 @@
+package io.jenkins.plugins.casc.impl.secrets;
+
+import hudson.Extension;
+import io.jenkins.plugins.casc.SecretSource;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * This {@link SecretSource} implementation allows to use a .properties file for providing secrets.
+ * The default file path is {@code /run/secrets/secrets.properties}, which can be changed via
+ * {@code SECRETS} environment variable.
+ *
+ * @author <a href="mailto:d.estermann.de@gmail.com">Daniel Estermann</a>
+ * @since 1.33
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+public class PropertiesSecretSource extends SecretSource {
+
+    private static final Logger LOGGER = Logger.getLogger(PropertiesSecretSource.class.getName());
+
+    /**
+     * Default path for .properties file
+     */
+    public static final String SECRETS_DEFAULT_PATH = "/run/secrets/secrets.properties";
+
+    private Properties secrets;
+
+    @Override
+    public Optional<String> reveal(String secret) {
+        // lazy initialization
+        if (secrets == null) {
+            secrets = new Properties();
+            final String secretsEnv = System.getenv("SECRETS");
+            final String secretsPath = secretsEnv == null ? SECRETS_DEFAULT_PATH : secretsEnv;
+            try (InputStream input = new FileInputStream(secretsPath)) {
+                secrets.load(input);
+            } catch (FileNotFoundException fnfe) {
+                LOGGER.log(Level.WARNING,
```

Comment:
I don't like this being a warning as I'm not using this feature why should I care that the file doesn't exist?

I think one of two things should be done:
1. just check if the file exists before trying to load?
2. move the log level down to FINE or reword it to say this is normal and have it at INFO?

Unless I'm misunderstanding how this works

Reply:
Reveal secrets should have no side effects.